proposals/0082-relax-transaction-constraints.md

```diff
@@ -0,0 +1,284 @@
+    regardless of if they are used in any instructions.
+15. The total loaded data size of the transaction must not exceed
+    `requested_loaded_accounts_data_size_limit`, or the default limit (64MB).
+16. Any account used as a program in a top-level instruction must:
+    - be the native loader: `NativeLoader1111111111111111111111111111111`
+    - OR
+      - exist
+      - be executable
+      - be owned by the native loader: `NativeLoader1111111111111111111111111111111`
+    - OR
+      - exist
+      - be executable
+      - the owner account be owned by the native loader: `NativeLoader1111111111111111111111111111111`
+      - the owner account must be executable
+
+### Proposed Protocol-Violation Errors
+
+The proposal is to move some items of the above list from protocol violations
+to runtime errors.
+Specifically, the following constraints will remain as protocol violations:
+
+1. Transaction must be deserializable via `bincode` (or equivalent) into a structure:
+
+    ```rust
+    pub struct VersionedTransaction {
+        #[serde(with = "short_vec")]
+        pub signatures: Vec<Signature>,
+        pub message: VersionedMessage,
+    }
+    ```
+
+    where `Signature` and `VersionedMessage` are defined in `solana-sdk`.
+2. Transaction serialized side must be less than or equal to 1232 bytes.
+3. Transaction signatures must be valid and in the same order as the static
+   account keys in the `VersionedMessage`.
+4. Transaction must have exactly the number of required signatures from the
+   `VersionedMessage` header.
+5. Transaction must not have more signatures than static account keys.
+6. The `VersionedMessage` must pass sanitization checks:
+    - The sum of the number of required signatures and the number of read-only
+      unsigned accounts must not be greater than the number of static account
+      keys.
+    - The number of readonly signed accounts must be less than the number of
+      required signatures.
+    - Each lookup table, if present, must be used to load at least one account.
+    - The total number of accounts, static or dynamic, must be less than 256.
+    - Each instruction's `program_id_index` must be less than the number of
+      static account keys.
+    - Each instruction's `program_id_index` must not be the payer index (0).
+    - All account indices in instructions must be less than the number of total
+      accounts.
+7. Transactions that use address lookup tables must be resolvable:
+    - The address lookup table account must exist.
+    - The address lookup table account must be owned by the address lookup
+      table program: `AddressLookupTab1e1111111111111111111111111`
+    - The address lookup table account data must be deserializable into
+      `AddressLookupTable` as defined in `solana-sdk`.
+    - All account table indices specified in the transaction must be less than
+      the number of active addresses in the address lookup table.
```

Comment:
Plan was for this to be in a separate follow-up SIMD since the choice here is very much not obvious imo.

How do we apply CUs to account-level cost if we cannot resolve ALTs?

Reply:
I'm fine with it being covered in a separate SIMD